Lib/asyncio/runners.py

```diff
@@ -130,6 +135,7 @@ def _lazy_init(self):
             return
         if self._loop_factory is None:
             self._loop = events.new_event_loop()
+            self._set_event_loop = True
```

Comment:
This is clever, but it seems unexpected that changing the loop factory would change the way set_event_loop is called

Reply:
Well currently if loop factory was not supplied then it calls `events.new_event_loop()` which initializes the policy registry hence we must call `set_event_loop`. However if the user provides a loop factory then we don't have set the event loop.

Reply:
Marking as resolved